src/Driver/PgSQL/Driver.php

```diff
@@ -64,9 +65,18 @@ private function constructConnectionString(
         #[SensitiveParameter]
         array $params
     ): string {
+        // pg_connect used by Doctrine DBAL does not support [...] notation,
+        // but requires the host address in plain form like `aa:bb:99...`
+        $matches = [];
+        if (preg_match('/^\[(.+)\]$/', $params['host'] ?? '', $matches) === 1) {
```

Comment:
```suggestion
        if (isset($params['host']) && preg_match('/^\[(.+)\]$/', $params['host'], $matches) === 1) {
```

No need to run a preg_match if there is no host.

Reply:
Yes makes sense!